opentelemetry-sdk/src/opentelemetry/sdk/trace/__init__.py

```diff
@@ -259,14 +259,15 @@ def end(self, end_time: Optional[int] = None) -> None:
                 raise RuntimeError("Calling end() on a not started span.")
             has_ended = self.end_time is not None
             if not has_ended:
+                if self.status is None:
+                    self.status = Status(canonical_code=StatusCanonicalCode.OK)
+
                 self.end_time = end_time if end_time is not None else time_ns()
+
         if has_ended:
```

Comment:
should this check be moved up as well?

Reply:
Not sure, should that check be protected by the lock too? Will have to think about it.

> It's not directly addressing to the PR but I think it's better to consider these improvements:
> 
> 1. Enable logging for tests. For example we can add
> 
> ```
> [pytest]
> log_cli = true
> log_cli_level = warning
> ```
> 
> 1. Wrap tests when we expect logs with something like this
> 
> ```
> ...
> with self.assertLogs(...):
>     # call it a second time
>     root.end()
>     ...
> ```
> 
> So we wouldn't spam in stdout about every log, but it can show unexpected warnings in code that mustn't produce warnings, critical, etc.

Reply:
I think it was outside of the lock originally, I'll look into this.

Reply:
I have noticed a few unusual things regarding the things we have in `self._lock` (like [this](https://github.com/open-telemetry/opentelemetry-python/pull/358/files#diff-2f7b55162ad705df55ee48eb71914737L280-L281)). I suggest we handle them in a separate PR.